### PR TITLE
Feat/pro 5075/aktualizr secondary storage

### PR DESCRIPTION
**Pull request by lbonn**
Based on #638 (will be rebased)

I am not entirely satisfied with the fact that it currently reuses the whole `UptaneConfig` from the primary, which contains a lot of irrelevant stuff (we need it to instantiate KeyManager) + badly named fields for the secondary.

Since `KeyManager` only uses a small portion of the content of `UptaneConfig`, I'll try to decouple them more in this PR. It can still be reviewed in the meantime.

### CODE REVIEW

**Comment by lbonn** [comment]
Got rid of `UptaneConfig` in `AktualizrSecondaryConfig`

**Comment by lbonn** [comment]
Rebased + a bug fix but don't know what's going on with Travis...

**Comment by lbonn** [comment]
The Travis tests passed after a couple of restarts...
It seems that it failed to pull some docker images for a while.

File: src/aktualizr_secondary/aktualizr_secondary.cc [hunk 1]
```diff
@@ -1,50 +1,80 @@
...
+// Initialize the parts needed for a secondary
+// - ECU keys
+// - serial id
+bool AktualizrSecondary::uptaneInitialize() {
+  if (keys_.generateUptaneKeyPair().size() == 0) {
+    return false;
+  }
+
+  // from uptane/initialize.cc but we only take care of our own serial/hwid
+  std::vector<std::pair<std::string, std::string> > ecu_serial;
```
Comment (patriotyk) [inline src/aktualizr_secondary/aktualizr_secondary.cc]: Maybe ecu_serials?

File: src/keymanager.h [hunk 2]
```diff
@@ -4,20 +4,30 @@
...
 #include "invstorage.h"
-#ifdef BUILD_P11
 #include "p11engine.h"
-#endif
+#include "types.h"
 #include "utils.h"
 
+// bundle some parts of the main config together
+struct KeyManagerConfig {
+  KeyManagerConfig() = delete;
```
Comment (patriotyk) [inline src/keymanager.h]: If thre is no any other constructors how it is possible to instantiate this struct?

File: tests/aktualizr_secondary/config_tests.toml [hunk 3]
```diff
@@ -1,2 +1,6 @@
 [network]
 port = 9031
+
+[storage]
+type = sqlite
```
Comment (patriotyk) [inline tests/aktualizr_secondary/config_tests.toml]: forgot double quotes

File: tests/aktualizr_secondary/config_tests.toml [hunk 4]
```diff
@@ -1,2 +1,6 @@
...
+schemas_path = config/schemas
```
Comment (patriotyk) [inline tests/aktualizr_secondary/config_tests.toml]: there too

Comment (lbonn) [reply tests/aktualizr_secondary/config_tests.toml]: thanks!

File: src/keymanager.h (hunk 2 above)
Comment (lbonn) [reply src/keymanager.h]: It should be derived from one of the main configs: `Config` and `AktualizrSecondaryConfig`. I didn't want it to have default value. So the way to construct it is `KeyManagerConfig kmconfig{field_values,...}`.

But yes, deleting the default constructor is not strictly needed and can be confusing. Maybe I'll replace it with a comment.

**Comment by cajun-rat** [comment]
With Serhiy's comments fixed, this looks good to me. Merging.

**Pull request merged**